crates/cubecl-opt/src/analyses/const_len.rs

```diff
@@ -0,0 +1,96 @@
+use std::{collections::HashMap, ops::Deref};
+
+use cubecl_core::ir::{Id, Operation, Operator, Variable, VariableKind};
+
+use crate::Optimizer;
+
+use super::Analysis;
+
+#[derive(Debug, Clone)]
+pub struct Slice {
+    pub start: Variable,
+    pub end: Variable,
+    pub end_op: Option<Operation>,
+    pub const_len: Option<u32>,
+}
```

Comment:
If I understand correctly, the pub analysis type is actually `Slices`, is there a reason to have `Slice` public?